envoy/server/factory_context.h

```diff
@@ -322,13 +294,7 @@ class FilterChainBaseAction : public Matcher::Action {
  * An implementation of FactoryContext. The life time should cover the lifetime of the filter chains
  * and connections. It can be used to create ListenerFilterChain.
  */
-class ListenerFactoryContext : public virtual FactoryContext {
-public:
-  /**
-   * Give access to the listener configuration
-   */
-  virtual const Network::ListenerConfig& listenerConfig() const PURE;
-};
+class ListenerFactoryContext : public virtual FactoryContext {};
```

Comment:
ooc if we keep this and it has no functions can it just be typeddefd to FactoryContext or does that cause inheritance problems?

Reply:
It's still painful with mocks/fakes. I thought it's better to delete it then - do you want to try that?

Reply:
I think we can just clean up it with a new PR.